src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1129,6 +1129,15 @@ private uint getMethodAttribsInternal(MethodDesc method)
             {
                 result |= CorInfoFlag.CORINFO_FLG_FORCEINLINE;
             }
+#if !READYTORUN
+            else if (method.OwningType is EcmaType ecmaOwningType
+                && ecmaOwningType.EcmaModule.EntryPoint == method)
+            {
+                // Mark entrypoint as NoInlining for better debugging. We don't want it to inline
+                // into startup code because breakpoints won't hit and stack traces won't have Main.
```

Comment:
If we want to the stacktraces to always have Main, we should also disable tailcalls from Main in `canTailCall`  JIT/EE method.

BTW: JIT does that too - https://github.com/dotnet/runtime/blob/021ecb8eb7998bf3288ee68b12e952f775828c10/src/coreclr/vm/jitinterface.cpp#L8229-L8237